File: bindings/lua/Polycode/SceneEntityInstance.lua
require "Polycode/Entity"

class "SceneEntityInstance" (Entity)


function SceneEntityInstance:__getvar(name)
	if name == "cloneUsingReload" then
		return Polycode.SceneEntityInstance_get_cloneUsingReload(self.__ptr)
	elseif name == "fileName" then
		return Polycode.SceneEntityInstance_get_fileName(self.__ptr)
	end
	if Entity["__getvar"] ~= nil then
		return Entity.__getvar(self, name)
	end
end

function SceneEntityInstance:__setvar(name,value)
	if name == "cloneUsingReload" then
		Polycode.SceneEntityInstance_set_cloneUsingReload(self.__ptr, value)
		return true
	elseif name == "fileName" then
		Polycode.SceneEntityInstance_set_fileName(self.__ptr, value)
		return true
	end
	if Entity["__setvar"] ~= nil then
		return Entity.__setvar(self, name, value)
	else
		return false
	end
end
function SceneEntityInstance:reloadEntityInstance()
	local retVal =  Polycode.SceneEntityInstance_reloadEntityInstance(self.__ptr)
end

function SceneEntityInstance:clearInstance()
	local retVal =  Polycode.SceneEntityInstance_clearInstance(self.__ptr)
end

function SceneEntityInstance:loadFromFile(fileName)
	local retVal = Polycode.SceneEntityInstance_loadFromFile(self.__ptr, fileName)
	return retVal
end

function SceneEntityInstance:getNumLinkedResourePools()
	local retVal =  Polycode.SceneEntityInstance_getNumLinkedResourePools(self.__ptr)
	return retVal
end

function SceneEntityInstance:getResourceEntry()
	local retVal =  Polycode.SceneEntityInstance_getResourceEntry(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<SceneEntityInstanceResourceEntry>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SceneEntityInstance:hasLayerID(layerID)
	local retVal = Polycode.SceneEntityInstance_hasLayerID(self.__ptr, layerID.__ptr)
	return retVal
end

function SceneEntityInstance:getNumLayers()
	local retVal =  Polycode.SceneEntityInstance_getNumLayers(self.__ptr)
	return retVal
end

function SceneEntityInstance:getFileName()
	local retVal =  Polycode.SceneEntityInstance_getFileName(self.__ptr)
	return retVal
end

function SceneEntityInstance:__delete()
	if self then Polycode.delete_SceneEntityInstance(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneEntityInstance.js
require('Polycode/Entity')

function SceneEntityInstance(fileName) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneEntityInstance(fileName)
	}
	Object.defineProperties(this, {
		'cloneUsingReload': { enumerable: true, configurable: true, get: SceneEntityInstance.prototype.__get_cloneUsingReload, set: SceneEntityInstance.prototype.__set_cloneUsingReload},
		'fileName': { enumerable: true, configurable: true, get: SceneEntityInstance.prototype.__get_fileName, set: SceneEntityInstance.prototype.__set_fileName}
	})
}


SceneEntityInstance.prototype = Object.create(Entity.prototype)

SceneEntityInstance.prototype.__get_cloneUsingReload = function() {
	return Polycode.SceneEntityInstance__get_cloneUsingReload(this.__ptr)
}

SceneEntityInstance.prototype.__set_cloneUsingReload = function(val) {
	Polycode.SceneEntityInstance__set_cloneUsingReload(this.__ptr, val)
}

SceneEntityInstance.prototype.__get_fileName = function() {
	return Polycode.SceneEntityInstance__get_fileName(this.__ptr)
}

SceneEntityInstance.prototype.__set_fileName = function(val) {
	Polycode.SceneEntityInstance__set_fileName(this.__ptr, val)
}


SceneEntityInstance.prototype.reloadEntityInstance = function() {
	Polycode.SceneEntityInstance_reloadEntityInstance(this.__ptr)
}

SceneEntityInstance.prototype.clearInstance = function() {
	Polycode.SceneEntityInstance_clearInstance(this.__ptr)
}

SceneEntityInstance.prototype.loadFromFile = function(fileName) {
	return Polycode.SceneEntityInstance_loadFromFile(this.__ptr, fileName)
}

SceneEntityInstance.prototype.getNumLinkedResourePools = function() {
	return Polycode.SceneEntityInstance_getNumLinkedResourePools(this.__ptr)
}

SceneEntityInstance.prototype.getResourceEntry = function() {
	var retVal = new SceneEntityInstanceResourceEntry("__skip_ptr__")
	retVal.__ptr = Polycode.SceneEntityInstance_getResourceEntry(this.__ptr)
	return retVal
}

SceneEntityInstance.prototype.hasLayerID = function(layerID) {
	return Polycode.SceneEntityInstance_hasLayerID(this.__ptr, layerID)
}

SceneEntityInstance.prototype.getNumLayers = function() {
	return Polycode.SceneEntityInstance_getNumLayers(this.__ptr)
}

SceneEntityInstance.prototype.getFileName = function() {
	return Polycode.SceneEntityInstance_getFileName(this.__ptr)
}
